File: airesources/Java/hlt/DebugLog.java
package hlt;

import java.io.FileWriter;
import java.io.IOException;

public class DebugLog {

    private final FileWriter file;
    private static DebugLog instance;

    private DebugLog(FileWriter f) {
        file = f;
    }

    static void initialize(FileWriter f) {
        instance = new DebugLog(f);
    }

    public static void addLog(String message) {
        try {
            instance.file.write(message);
            instance.file.write('\n');
            instance.file.flush();
        }
        catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: airesources/Scala/hlt/DebugLog.scala
package hlt

import java.io.{FileWriter, IOException}

object DebugLog {
  private var writer: FileWriter = _

  def addLog(message: String): Unit =
    try {
      writer.write(message)
      writer.write('\n')
      writer.flush()
    } catch {
      case e: IOException =>
        e.printStackTrace()
    }

  private[hlt] def initialize(f: FileWriter): Unit = writer = f
}
